classes/local/manager.php

```diff
@@ -139,22 +140,28 @@ public static function update_object_by_hash($contenthash, $newlocation) {
             $newobject->timeduplicated = $oldobject->timeduplicated;
             $newobject->id = $oldobject->id;
 
-            // If location hasn't changed we do not need to update.
-            if ((int)$oldobject->location === $newlocation) {
+            // If location hasn't changed we do not need to update unless filesize is not populated.
+            if ((int)$oldobject->location === $newlocation && !empty($oldobject->filesize)) {
```

Comment:
similar to above I think we should not use empty we should use === null everywhere